examples/misc/test/language_tour/classes_test.dart

```diff
@@ -79,6 +84,28 @@ void main() {
     expect(_test, m.prints('The type of a is ImmutablePoint'));
   });
 
+  test('const_context', () {
+    // #docregion const-context-withconst
+    // Lots of const keywords here.
+    const pointAndLine1 = const {
+      'point': const [const ImmutablePoint(0, 0)],
+      'line': const [const ImmutablePoint(1, 10), const ImmutablePoint(-2, 11)],
+    };
+    // #enddocregion const-context-withconst
+
+    // #docregion const-context-noconst
+    // Only one const, which establishes the constant context.
+    const pointAndLine2 = {
+      'point': [ImmutablePoint(0, 0)],
+      'line': [ImmutablePoint(1, 10), ImmutablePoint(-2, 11)],
+    };
+    // #enddocregion const-context-noconst
+
+    // Same instances!
+    assert(identical(pointAndLine1, pointAndLine2));
```

Comment:
Suggestions:
- Use `expect()`
- `const` values are canonicalized, so you can use `==`

E.g.,

```dart
expect(pointAndLine1 == pointAndLine2, isTrue);
...
```